Apps/Src/DebugDump.c

```diff
@@ -0,0 +1,141 @@
+extern os_error_loc_t OSErrLocBitmap;
+extern bool UpdateVel_ToggleCruise;
+extern uint16_t SupplementalVoltage;
+extern uint32_t StateOfCharge;
+
+static const char *MINIONPIN_STRING[] = {
+    FOREACH_MinionPin(GENERATE_STRING)
+};
+
+static const char *CONTACTOR_STRING[] = {
+    FOREACH_contactor(GENERATE_STRING)
+};
+
+// Need to keep this in sync with Task.h
+/*----------------------------------------------*/
+#define FAULT_BITMAP_NUM 6
+#define OS_LOC_NUM 14
+static const char *OS_LOC_STRING[] = { 
+    "OS_NONE_LOC", 
+    "OS_ARRAY_LOC", 
+    "OS_READ_CAN_LOC", 
+    "OS_READ_TRITIUM_LOC",
+    "OS_SEND_CAN_LOC",
+    "OS_SEND_TRITIUM_LOC",
+    "OS_UPDATE_VEL_LOC",
+    "OS_CONTACTOR_LOC",
+    "OS_MINIONS_LOC",
+    "OS_MAIN_LOC",
+    "OS_CANDRIVER_LOC",
+    "OS_MOTOR_CONNECTION_LOC",
+    "OS_DISPLAY_LOC"
+};
+
+static const char *FAULT_BITMAP_STRING[] = { 
+    "FAULT_NONE", 
+    "FAULT_OS", 
+    "FAULT_UNREACH", 
+    "FAULT_TRITIUM",
+    "FAULT_READBPS",
+    "FAULT_DISPLAY",
+    "FAULT_BPS"
+};
+/*----------------------------------------------*/
+
+void Task_DebugDump(void* p_arg) {
+    OS_ERR err;
+    Minion_Error_t mErr;
+
+    while(1){
+
+        // Get pedal information
+        int8_t accelPedal = Pedals_Read(ACCELERATOR);
+        printf("ACCELERATOR: %d\n\r", accelPedal);
+
+        int8_t brakePedal = Pedals_Read(BRAKE);
+        printf("BRAKE: %d\n\r", brakePedal);
+
+        // Get minion information
+        for(MinionPin_t pin = 0; pin < MINIONPIN_NUM-1; pin++){ // Plan to change MINIONPIN_NUM-1 -> MINIONPIN_NUM after Minion function is fixed
+            bool pinState = Minion_Read_Pin(pin, &mErr);
+            printf("%s: %s\n\r", MINIONPIN_STRING[pin], pinState == true ? "true" : "false");
+        }
+
+        // Get contactor info
+        for(contactor_t contactor = 0; contactor < CONTACTOR_NUM; contactor++){
+            bool contactorState = Contactors_Get(contactor) == ON ? true : false;
+            printf("%s: %s\n\r", CONTACTOR_STRING[contactor], contactorState == true ? "true" : "false");
+        } 
+
+        // Send Tritium variables
+        printf("Cruise Enable: %s\n\r", get_cruiseEnable() ? "true" : "false");
+        printf("Cruise Set: %s\n\r", get_cruiseSet() ? "true" : "false");
+        printf("One Pedal Enable: %s\n\r", get_onePedalEnable() ? "true" : "false");
+        printf("Regen Enable: %s\n\r", get_regenEnable() ? "true" : "false");
+        printf("Pedal Brake Percent: %d\n\r", get_brakePedalPercent());
+        printf("Pedal Accel Percent: %d\n\r", get_accelPedalPercent());
+        // printf("Pedal Accel Percent: %s\n\r", get_accelPedalPercent()); // Gear
+        // printf("Current Setpoint: %f", get_currentSetpoint());   // float print not working?
+        // printf("Current Setpoint: %f", get_currentSetpoint());
+        // printf("Current Setpoint: %f", get_currentSetpoint());
+        // printf("Current Setpoint: %f", get_currentSetpoint());
```

Comment:
There's a print_float function in SendTritium. We can move that to somewhere and use it.